credentials/java/com/google/auth/Credentials.java

```diff
@@ -77,4 +106,20 @@
    * @throws IOException if there was an error getting up-to-date access.
    */
   public abstract void refresh() throws IOException;
+
+  /**
+   * The callback that receives the result of the asynchronous {@link #getRequestMetadata(URI,
+   * Executor, Callback)}. Exactly one method should be called.
+   */
+  public interface Callback {
```

Comment:
I'm okay with using RequestMetadataCallback. I would generally prefer the inner class (it isn't nested, since it is static) in this case, but it's also bikeshedding. Given how few things are in the com.google.auth package, I expect it doesn't matter much one way or another here.